operator/pkg/s3.go

```diff
@@ -5,25 +5,56 @@ import (
 	"fmt"
 	"io"
 	"net/http"
+	"time"
 
 	"github.com/ugorji/go/codec"
 
 	"github.com/yetanotherco/aligned_layer/operator/merkle_tree"
 )
 
-func (o *Operator) getBatchFromS3(ctx context.Context, batchURL string, expectedMerkleRoot [32]byte) ([]VerificationData, error) {
+func (o *Operator) getBatchFromS3(ctx context.Context, batchURL string, expectedMerkleRoot [32]byte, maxRetries int, retryDelay time.Duration) ([]VerificationData, error) {
 	o.Logger.Infof("Getting batch from S3..., batchURL: %s", batchURL)
 
-	req, err := http.NewRequestWithContext(ctx, "GET", batchURL, nil)
+	var resp *http.Response
+	var err error
+	var req *http.Request
+
+	for attempt := 0; attempt < maxRetries; attempt++ {
+		if attempt > 0 {
+			o.Logger.Infof("Retrying S3 fetch (attempt %d of %d)", attempt+1, maxRetries)
```

Comment:
nice to have, log how long will wait before retry
```suggestion
			o.Logger.Infof("Waiting for %s before retrying S3 fetch (attempt %d of %d)", retryDelay, attempt+1, maxRetries)
```